django/core/management/commands/makemessages.py

```diff
@@ -349,11 +349,14 @@ def handle(self, *args, **options):
             self.default_locale_path = self.locale_paths[0]
             self.invoked_for_django = True
         else:
+            # Settings are available if run from project root
             if self.settings_available:
                 self.locale_paths.extend(settings.LOCALE_PATHS)
-            # Allow to run makemessages inside an app dir
-            if os.path.isdir('locale'):
+            # if settings aren't available, or locale path exists we use the locale path
```

Comment:
Django style is to avoid the use of _we_.
Also, since the variable name already expresses whether settings are available, I would find it clearer to keep the remark about detecting that the command was run from the app directory.

Reply:
yeah, this conditional is technically there to support the way the tests are structured which is what made it confusing to me. Technically, this statement shouldn't be needed, and it should be enough to just check if the settings aren't available but since the tests folder has a `locale` folder, it becomes needed to add it to the path. Regardless, I don't think it hurts to have this check anyways right?

Reply:
Of course, and actually I was referring to the comment only, not the conditional.